Kerberos.NET/Entities/Krb/KrbKdcRep.cs

```diff
@@ -102,7 +105,12 @@ public static T GenerateServiceTicket<T>(ServiceTicketRequest request)
 
             var authz = GenerateAuthorizationData(request);
 
-            var sessionKey = KrbEncryptionKey.Generate(request.PreferredClientEType ?? request.ServicePrincipalKey.EncryptionType);
+            var sessionKey = encryptionKey;
```

Comment:
I would change the method parameter name to be `sessionKey` to be explicit about what this is. Then you don't need a local variable for it either, just the `if`.